runtime/near-vm-runner/src/preload.rs

```diff
@@ -12,13 +12,25 @@ use near_vm_logic::types::PromiseResult;
 use near_vm_logic::{External, ProtocolVersion, VMConfig, VMContext, VMKind, VMOutcome};
 
 use crate::cache;
+use crate::memory::WasmerMemory;
 use crate::preload::VMModule::{Wasmer0, Wasmer1};
-use crate::wasmer1_runner::{default_wasmer1_store, run_wasmer1_module};
+use crate::wasmer1_runner::{default_wasmer1_store, run_wasmer1_module, Wasmer1Memory};
 use crate::wasmer_runner::run_wasmer0_module;
 
 enum VMModule {
     Wasmer0(wasmer_runtime::Module),
-    Wasmer1(wasmer::Module, wasmer::Store),
+    Wasmer1(wasmer::Module),
+}
+
+enum VMDataPrivate {
+    Wasmer0(WasmerMemory),
+    Wasmer1(Wasmer1Memory),
+}
+
+#[derive(Clone)]
+enum VMDataShared {
+    Wasmer0,
+    Wasmer1(wasmer::Store),
```

Comment:
I feel it can be problematic to share store between instances. From this doc: https://docs.rs/wasmer/1.0.2/wasmer/struct.Store.html:

> The store represents all global state that can be manipulated by WebAssembly programs. It consists of the runtime representation of all instances of functions, tables, memories, and globals that have been allocated during the lifetime of the abstract machine.

Would there be name conflictions, and if not clean up store in each run, could it grow infinitely?

Reply:
Note that intended usage of `ContractCaller` is the timespan of one block, it shall not be reused between blocks. What shall be reused need to be decided based on actual perf data, agree with that. Current infrastructure allows rather flexible selection of what is shared and what is recreated, that's the point of `vm_data_shared` and `vm_data_private`.

Reply:
@olonho do we cache any `ContractCallPrepareResult`?

Reply:
No, it's pretty transient, like handle.